spec/unit/functions/new_spec.rb

```diff
@@ -362,6 +390,34 @@
       MANIFEST
       )).to have_resource('Notify[Integer, 42]')
     end
+
+    it "produces an absolute value when second argument is 'true'" do
+      expect(eval_and_collect_notices(<<-MANIFEST
+        notice(Numeric.new(-42.3, true))
+      MANIFEST
+      )).to eql(['42.3'])
+    end
+
+    it "does not produce an absolute value when second argument is 'false'" do
+      expect(eval_and_collect_notices(<<-MANIFEST
+        notice(Numeric.new(-42.3, false))
+      MANIFEST
+      )).to eql(['-42.3'])
+    end
+
+    it "produces an absolute value from hash {from => val, abs => true}" do
+      expect(eval_and_collect_notices(<<-MANIFEST
+        notice(Numeric.new({from => -42.3, abs => true}))
+      MANIFEST
+      )).to eql(['42.3'])
+    end
+
+    it "does not produce an absolute value from hash {from => val, abs => false}" do
+      expect(eval_and_collect_notices(<<-MANIFEST
+        notice(Numeric.new({from => -42.3, abs => false}))
+      MANIFEST
+      )).to eql(['-42.3'])
+    end
```

Comment:
The added examples (4 of them) use a Float value, but are in the context of testing Integer.